From 2c0428f73eddda137e8aa6ef4cb0112b64f0e2d3 Mon Sep 17 00:00:00 2001
From: Thomas White <tomwhite@apache.org>
Date: Wed, 27 Feb 2013 10:42:58 +0000
Subject: [PATCH 1301/1357] MR1: MAPREDUCE-5008. Merger progress miscounts with respect to EOF_MARKER. Contributed by Sandy Ryza.

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-1@1450725 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit f15a72d7d2d14e8d715e4404f2eb55e5e78e67e9)

Conflicts:

	src/mapred/org/apache/hadoop/mapred/ReduceTask.java

Author: Sandy Ryza
Reason: Fix progress after MAPREDUCE-2264
Ref: CDH-11368
---
 .../org/apache/hadoop/mapred/ReduceTask.java       |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/ReduceTask.java b/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/ReduceTask.java
index aa8d541..3c0f83c 100644
--- a/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/ReduceTask.java
+++ b/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/ReduceTask.java
@@ -2472,8 +2472,8 @@ public class ReduceTask extends Task {
               keyClass, valueClass, reduceTask.codec, null);
           try {
             Merger.writeFile(rIter, writer, reporter, conf);
-            long decompressedBytesWritten = writer.decompressedBytesWritten;
             writer.close();
+            long decompressedBytesWritten = writer.decompressedBytesWritten;
             writer = null;
             FileStatus fileStatus = fs.getFileStatus(outputPath);
             CompressAwareFileStatus compressedFileStatus = new CompressAwareFileStatus(
@@ -2716,8 +2716,8 @@ public class ReduceTask extends Task {
                                   reduceTask.spilledRecordsCounter, null);
               
               Merger.writeFile(iter, writer, reporter, conf);
-              decompressedBytesWritten = writer.decompressedBytesWritten;
               writer.close();
+              decompressedBytesWritten = writer.decompressedBytesWritten;
             } catch (Exception e) {
               localFileSys.delete(outputPath, true);
               throw new IOException (StringUtils.stringifyException(e));
@@ -2831,8 +2831,8 @@ public class ReduceTask extends Task {
             combineCollector.setWriter(writer);
             combinerRunner.combine(rIter, combineCollector);
           }
-          decompressedBytesWritten = writer.decompressedBytesWritten;
           writer.close();
+          decompressedBytesWritten = writer.decompressedBytesWritten;
 
           LOG.info(reduceTask.getTaskID() + 
               " Merge of the " + noInMemorySegments +
-- 
1.7.0.4

